sql/wsrep_sst.cc

```diff
@@ -377,6 +392,67 @@ bool wsrep_sst_continue ()
   return false;
 }
 
+// Pointer to SST process object:
+
+static wsp::process * sst_process= static_cast<wsp::process*>(NULL);
+
+static void wsrep_clear_process (void)
+{
+  // Clear the pointer to SST process:
+  if (mysql_mutex_lock (&LOCK_wsrep_sst)) abort();
+  sst_process = NULL;
+  mysql_mutex_unlock (&LOCK_wsrep_sst);
+}
+
+// SST cancellation flag:
+
+static bool sst_cancelled= false;
+
+void wsrep_sst_cancel (bool call_wsrep_cb)
+{
+  if (mysql_mutex_lock (&LOCK_wsrep_sst)) abort();
+  if (!sst_cancelled)
+  {
+    sst_cancelled=true;
+    /*
+      When we launched the SST process, then we need
+      to terminate it before exit from the parent (server)
+      process:
+    */
+    if (sst_process)
+    {
+      sst_process->terminate();
+      sst_process = NULL;
+    }
+    /*
+      If this is a normal shutdown, then we need to notify
+      the wsrep provider about completion of the SST, to
+      prevent infinite waitng in the wsrep provider after
+      the SST process was canceled:
+    */
+    if (call_wsrep_cb && sst_awaiting_callback)
+    {
+      WSREP_INFO("Signalling cancellation of the SST request.");
+      wsrep_gtid_t const state_id = {
+          WSREP_UUID_UNDEFINED,
+          WSREP_SEQNO_UNDEFINED
+      };
+      sst_awaiting_callback = false;
+      wsrep->sst_received(wsrep, &state_id, NULL, 0, -ECANCELED);
+    }
+  }
+  mysql_mutex_unlock (&LOCK_wsrep_sst);
+}
```

Comment:
One more SST fix https://github.com/percona/percona-xtradb-cluster/commit/151f91e3b8bb - why it's here? I stop to review the SST changes - we need to discuss why they are here.